services/src/datasets/external/netcdfcf/database.rs

```diff
@@ -0,0 +1,1438 @@
+                tokio_postgres::types::Type::UUID,
+                tokio_postgres::types::Type::TEXT,
+                tokio_postgres::types::Type::INT8,
+            ],
+        )
+        .await
+        .boxed_context(error::UnexpectedExecution)?;
+
+    for time_step in time_steps {
+        transaction
+            .execute(
+                &timestamp_statement,
+                &[&provider_id, &file_name, &time_step],
+            )
+            .await
+            .boxed_context(error::UnexpectedExecution)?;
+    }
+
+    Ok(())
+}
+
+async fn insert_loading_infos(
+    transaction: &Transaction<'_>,
+    provider_id: &DataProviderId,
+    file_name: &str,
+    loading_infos: &[LoadingInfoMetadata],
+) -> Result<()> {
+    let loading_info_stmt = transaction
+        .prepare_typed(
+            r#"
+            INSERT INTO ebv_provider_loading_infos (
+                provider_id,
+                file_name,
+                group_names,
+                entity_id,
+                meta_data
+            ) VALUES (
+                $1,
+                $2,
+                $3 :: TEXT[],
+                $4,
+                $5 :: "GdalMetaDataList"
+            );"#,
+            &[
+                tokio_postgres::types::Type::UUID,
+                tokio_postgres::types::Type::TEXT,
+                tokio_postgres::types::Type::TEXT_ARRAY,
+                tokio_postgres::types::Type::INT8,
+                // we omit `GdalMetaDataList`
+            ],
+        )
+        .await
+        .boxed_context(error::CannotStoreLoadingInfo)?;
+
+    for loading_info in loading_infos {
+        transaction
+            .execute(
+                &loading_info_stmt,
+                &[
+                    &provider_id,
+                    &file_name,
+                    &loading_info.group_path,
+                    &(loading_info.entity_id as i64),
+                    &loading_info.meta_data,
+                ],
+            )
+            .await
+            .boxed_context(error::CannotStoreLoadingInfo)?;
+    }
+
+    Ok(())
+}
+
+async fn remove_overviews(
+    transaction: &Transaction<'_>,
+    provider_id: DataProviderId,
+    file_name: &str,
+) -> Result<bool> {
+    let removed_rows = transaction
+        .execute(
+            "
+            DELETE FROM
+                ebv_provider_overviews
+            WHERE
+                provider_id = $1 AND
+                file_name = $2
+            ",
+            &[&provider_id, &file_name],
+        )
+        .await
+        .boxed_context(error::UnexpectedExecution)?;
+
+    Ok(removed_rows > 0)
+}
+
+pub struct NetCdfDatabaseListingConfig<'a> {
+    pub provider_id: DataProviderId,
+    pub collection: &'a LayerCollectionId,
+    pub file_name: &'a str,
+    pub group_path: &'a [String],
+    pub options: &'a LayerCollectionListOptions,
+}
+
+// TODO: move
+#[cfg(feature = "pro")]
+#[async_trait]
+impl<Tls> NetCdfCfProviderDb for crate::pro::contexts::ProPostgresDb<Tls>
```

Comment:
is there any difference to free?

Reply:
[75304d5](https://github.com/geo-engine/geoengine/pull/941/commits/75304d5a77fd8a73b3ee7afe37f0427f38a1b935)